pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -9,8 +9,72 @@ let
     # to build it. This is a bit confusing for cross compilation.
     inherit (stdenv) hostPlatform;
   };
+
+  makeOverlayable = mkDerivationSimple:
+    fnOrAttrs:
+      if builtins.isFunction fnOrAttrs
```

Comment:
Perhaps `lib.isFunction` instead? This way functors will work.

Reply:
I've considered it, but
 - this is performance sensitive code
 - functors are for "objects" that double as a function, usually for compatibility reasons. I don't expect these to be used here
 - it is more of a breaking change, as someone may already be using this to turn a package into a functor, in which case the functor _must not_ be invoked
 - it doesn't add new functionality. An eta expansion of the functor is enough to make it work: `mkDerivation (finalAttrs: functorObj finalAttrs)`